src/client.js

```diff
@@ -107,13 +104,23 @@ class ShotgunApiClient {
 		return body;
 	}
 
+	async tokenExpired(token) {
+
+		if (!token)
+			return false;
+
+		let jwtPayload = JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString());
+
+		return (Date.now() / 1000) > jwtPayload['exp'];
+	}
+
 	async getAuthorizationHeader() {
 
-		let { token, tokenExpirationTimestamp } = this;
+		let { token } = this;
 
-		if (!token)
+		if (!token || await this.tokenExpired(token['refresh_token']))
```

Comment:
`await` is no longer needed due to above change.
```suggestion
		if (!token || this.tokenExpired(token.refresh_token))
```